src/Psalm/Internal/Analyzer/Statements/GlobalAnalyzer.php

```diff
@@ -79,6 +92,20 @@ public static function analyze(
                         $var,
                         $var_id
                     );
+
+                    if ($statements_analyzer->data_flow_graph instanceof VariableUseGraph) {
+                        if ($global_context !== null && $global_context->hasVariable($var_id)) {
+                            // TODO get global CodeLocation so that global variable can be marked as unused if it is
+                            // unused inside the function. Marking it as a referenced global causes it to be marked
+                            // used if it's used as a global in any function, even if it's unused in that function.
```

Comment:
I would love for this to work, but I've spent some time looking and I still have no idea how to find the statement where the global was declared in the global scope.
```
$a = 0; // How do I find this statement so that it can be marked as unused

function foo(): void {
    global $a; // when analyzing this statement
}
```

Reply:
This is basically not solvable:
- the statement could be in another file
- there could be multiple statement in multiple files with multiple types
- even if the statement is in the same file, how do you make sure you executed it when you called the function?
- there could be a side-effect from another function that could change the variable type in another function declared in another file

Global scope is a mess

Reply:
Yeah, that's what I was afraid of. Guess I'll take out the TODO and related stuff. 🙁